### PR TITLE
[BOLT][DWARF] Remove option to write to DWP

### PR DESCRIPTION
**Pull request by sayhaan**
Remove the --write-dwp option as well as related code and tests.

### CODE REVIEW

**Review by ayermolo — APPROVED** [review]

**Comment by ayermolo** [comment]
In emitDWOBuilder 
DWARFRewriter::UnitMetaVectorType TUMetaVector;
  DWARFRewriter::UnitMeta CUMI = {0, 0, 0};
  
 Needs to be cleaned up

**Review by ayermolo — CHANGES_REQUESTED** [review]
What about declaration of those data structures?

**Comment by sayhaan** [comment]
> What about declaration of those data structures?

About to push those changes, was fixing a rebase issue

**Comment by ayermolo** [comment]
> > What about declaration of those data structures?
> 
> About to push those changes, was fixing a rebase issue

Why was rebase necessary?

**Comment by sayhaan** [comment]
> > > What about declaration of those data structures?
> > 
> > 
> > About to push those changes, was fixing a rebase issue
> 
> Why was rebase necessary?

Was running git fetch upstream and git rebase upstream/main

**Comment by ayermolo** [comment]
> > > > What about declaration of those data structures?
> > > 
> > > 
> > > About to push those changes, was fixing a rebase issue
> > 
> > 
> > Why was rebase necessary?
> 
> Was running git fetch upstream and git rebase upstream/main

Were you getting rebase conflicts on this PR?

File: bolt/lib/Rewrite/DWARFRewriter.cpp [hunk 1]
```diff
@@ -470,23 +464,18 @@ createDIEStreamer(const Triple &TheTriple, raw_pwrite_stream &OutFile,
   return Streamer;
 }
 
-static DWARFRewriter::UnitMeta
-emitUnit(DIEBuilder &DIEBldr, DIEStreamer &Streamer, DWARFUnit &Unit) {
+void emitUnit(DIEBuilder &DIEBldr, DIEStreamer &Streamer, DWARFUnit &Unit) {
```
Comment (ayermolo) [inline bolt/lib/Rewrite/DWARFRewriter.cpp:467]: static was dropped

**Comment by sayhaan** [comment]
> > > > > What about declaration of those data structures?
> > > > 
> > > > 
> > > > About to push those changes, was fixing a rebase issue
> > > 
> > > 
> > > Why was rebase necessary?
> > 
> > 
> > Was running git fetch upstream and git rebase upstream/main
> 
> Were you getting rebase conflicts on this PR?

Yes, initially got the "error: failed to push some refs" message so I did fetch upstream/rebase. I was getting an error when I rebased too, so I aborted and reset from reflog and pushed again.

**Review by ayermolo — APPROVED** [review]

**Pull request merged**